src/plugins/intel_gpu/src/runtime/kernels_cache.cpp

```diff
@@ -562,10 +566,13 @@ void kernels_cache::load(BinaryInputBuffer& ib) {
     }
 }
 
-kernels_cache::compiled_kernels kernels_cache::compile(const kernel_impl_params& params,
+std::vector<kernel::ptr> kernels_cache::compile(const kernel_impl_params& params,
                                             const std::vector<std::shared_ptr<kernel_string>>& kernel_sources,
                                             bool dump_custom_program) {
-    OV_ITT_SCOPED_TASK(ov::intel_gpu::itt::domains::intel_gpu_plugin, "KernelsCache::Compile_ThreadSafe");
+    OV_ITT_SCOPED_TASK(ov::intel_gpu::itt::domains::intel_gpu_plugin, "KernelsCache::compile");
+    if (kernel_sources.empty())
```

Comment:
When can this happen?

Reply:
Investigate the case kernel_sources is empty and fix more code